cmd/gitserver/internal/git/gitcli/commitlog.go

```diff
@@ -159,12 +168,17 @@ func (it *commitLogIterator) Close() error {
 	if err != nil {
 		var e *CommandFailedError
 		if errors.As(err, &e) && e.ExitStatus == 128 {
-			if bytes.Contains(e.Stderr, []byte("not a tree object")) || bytes.Contains(e.Stderr, []byte("fatal: bad object")) {
-				return &gitdomain.RevisionNotFoundError{Repo: it.repoName, Spec: it.spec}
-			}
-			if bytes.Contains(e.Stderr, []byte("fatal: your current branch")) && bytes.Contains(e.Stderr, []byte("does not have any commits yet")) {
+			if (bytes.Contains(e.Stderr, []byte("fatal: your current branch")) && bytes.Contains(e.Stderr, []byte("does not have any commits yet"))) || bytes.Contains(e.Stderr, []byte("fatal: bad revision 'HEAD'")) {
 				return nil
 			}
+
+			if bytes.Contains(e.Stderr, []byte("not a tree object")) ||
```

Comment:
I think this would be easier to read if this was all extracted to a for loop

```suggestion
var errMessages = []string{
	"not a tree object",
	"fatal: bad object",
	"fatal: ambiguous argument",
	"unknown revision or path not in the working tree.",
	"fatal: Invalid revision range",
	"fatal: bad revision",
}

for _, message := range errMessages {
	if bytes.Contains(e.Stderr, []byte(message)) {
		return &gitdomain.RevisionNotFoundError{Repo: it.repoName, Spec: it.spec}
	}
}
```

Reply:
nice, thanks!